### PR TITLE
Cat 1491 yoi recategorisation category decision page updates

### PR DESCRIPTION
**Pull request by Olusola-Opencast**

### CODE REVIEW

File: server/views/formPages/recat/decision.html [hunk 1]
```diff
@@ -80,59 +79,45 @@ <h1 id="securityInputHeader" class="govuk-heading-l">{{ pageTitle }}</h1>
...
+  {% endif %}
 
-{% if user.activeCaseLoad.female %}
   {{ govukRadios({
     name: "category",
     classes: "govuk-radios",
     fieldset: {
       legend: {
-        text: "Which category is most suitable for this person?",
-        isPageHeading: false
```
Comment (kberec) [inline server/views/formPages/recat/decision.html:94]: Reminder to check

File: server/views/formPages/supervisor/recatReview.html [hunk 2]
```diff
@@ -109,32 +112,39 @@ <h1 class="govuk-heading-l openConditionsHeader">Open Conditions</h1>
...
+        {% set noLabel = 'No' %}
+      {% else  %}
+        {% set radios = radios | rejectattr('B') | rejectattr('C') | rejectattr('D')%}
+        {% set hiddenRadioClasses = ' govuk-radios__conditional--hidden' %}
+        {% set hiddenRadioLegendClasses = ' govuk-radios__conditional--hidden' %}
+        {% if (currentCat == "R") %}
+          {% set noLabel = "No, consider for open" %}
+        {% elif (currentCat == "T") %}
+          {% set noLabel = "No, closed is more appropriate" %}
+        {% endif %}
```
Comment (a6software) [inline server/views/formPages/supervisor/recatReview.html:131]: Only a suggestion:

```
    {% if isFemale %}
      {% set radios = radios | rejectattr('B') | rejectattr('C') | rejectattr('D') %}
      {% set noLabel = 'No' %}

      {% if not data.details.youngOffender %}
        {% set hiddenRadioClasses = ' govuk-radios__conditional--hidden' %}
        {% set hiddenRadioLegendClasses = ' govuk-radios__conditional--hidden' %}
        {% if (currentCat == "R") %}
          {% set noLabel = "No, consider for open" %}
        {% elif (currentCat == "T") %}
          {% set noLabel = "No, closed is more appropriate" %}
        {% endif %}
      {% endif %}
    {% else %}
      {% set radios = radios | rejectattr('R') | rejectattr('T') %}
      {% set noLabel = 'No' %}
    {% endif %}
```

Or to de-nest by one layer:

```
    {% set isYoungOffender = data.details.youngOffender %}
    {% if isFemale and isYoungOffender %}
      {% set radios = radios | rejectattr('B') | rejectattr('C') | rejectattr('D') %}
      {% set noLabel = 'No' %}
    {% elif isFemale and not isYoungOffender %}
      {% set radios = radios | rejectattr('B') | rejectattr('C') | rejectattr('D')%}
      {% set hiddenRadioClasses = ' govuk-radios__conditional--hidden' %}
      {% set hiddenRadioLegendClasses = ' govuk-radios__conditional--hidden' %}
      {% if (currentCat == "R") %}
        {% set noLabel = "No, consider for open" %}
      {% elif (currentCat == "T") %}
        {% set noLabel = "No, closed is more appropriate" %}
      {% endif %}
    {% else %}
      {% set radios = radios | rejectattr('R') | rejectattr('T') %}
      {% set noLabel = 'No' %}
    {% endif %}
```

File: server/views/formPages/recat/decision.html (hunk 1 above)
Comment (kberec) [reply server/views/formPages/recat/decision.html:94]: In prototype: `What security category is most suitable for this prisoner?`
Check config/recat.js and recat/DecisionSpecification too

File: server/views/formPages/recat/decision.html [hunk 3]
```diff
@@ -80,59 +79,45 @@ <h1 id="securityInputHeader" class="govuk-heading-l">{{ pageTitle }}</h1>
...
-
-{% else %}
-
-  {{ govukRadios({
-    name: "category",
-    classes: "govuk-radios",
-    fieldset: { legend: { text: "What security category is most suitable for this person?" } },
-    errorMessage: {
-      text: "Select what category is most suitable for this person"
+      text: "Select the category is most suitable for this person"
```
Comment (kberec) [inline server/views/formPages/recat/decision.html:118]: In the ticket: `Select the category that is most suitable for this person`

File: test/routes/recat.test.js [hunk 4]
```diff
@@ -185,79 +210,74 @@ describe('recat', () => {
...
+    return request(app)
+      .get(`/decision/12345`)
+      .expect(200)
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        expect(res.text).toContain('Category decision')
+        expect(res.text).not.toContain('catCOption')
+        expect(res.text).toContain('catIOption')
+        expect(res.text).toContain('openOption')
+        expect(res.text).not.toContain('Prisoner has an indeterminate sentence')
```
Comment (kberec) [inline test/routes/recat.test.js:272]: Can you please consider adding:
```
        expect(res.text).toContain('catJOption')
        expect(res.text).toContain('closedOption')
```

File: server/views/formPages/supervisor/recatReview.html [hunk 5]
```diff
@@ -79,15 +79,18 @@ <h1 class="govuk-heading-l openConditionsHeader">Open Conditions</h1>
       {
         value: "R",
         id: "overriddenCategoryR",
-        checked: true,
+        checked: data.supervisor.review.supervisorOverriddenCategory === 'R',
         text: "Consider for closed",
         R: true
       },
       {
         value: "T",
```
Comment (kberec) [inline server/views/formPages/supervisor/recatReview.html:87]: Please consider switching order of female open.closed with YOI open/closed to match the ticket.

File: server/views/formPages/recat/decision.html (hunk 1 above)
Comment (Olusola-Opencast) [reply server/views/formPages/recat/decision.html:94]: Just checked the git branches - on main it states "What category is most suitable for this person?"  On the prototype for women's estate option 2 it states "What category is most suitable for this person?"  at https://categorisation-prototype-main.apps.live.cloud-platform.service.justice.gov.uk/v1/cat/decision?cya=false.  I am aware that Lisa mentioned some content rules where you can't call an offender "prisoner" 
I will have to see if I can change the text on whether they are male or female.

Comment (kberec) [reply server/views/formPages/recat/decision.html:94]: I'd suggest to discuss it with Barry

File: server/views/formPages/supervisor/recatReview.html (hunk 5 above)
Comment (Olusola-Opencast) [reply server/views/formPages/supervisor/recatReview.html:87]: Don't understand there is no screenshot and there is nothing on the ticket stating this for the supervisor page.

**Review by kberec — APPROVED** [review]

**Pull request merged**